pkg/config/server.go

```diff
@@ -29,32 +30,64 @@ type UserRolesConfig struct {
 type UserConfig struct {
 	PublicKey string   `mapstructure:"public_key"`
 	Name      string   `mapstructure:"name"`
+	GithubID  string   `mapstructure:"github_id"`
 	Targets   []string `mapstructure:"targets"`
 }
 
 func (uc *UserConfig) ID() string {
 	return base64.StdEncoding.EncodeToString([]byte(uc.PublicKey))
 }
 
+type GithubConfig struct {
+	APIEndpoint string `mapstructure:"api_endpoint"`
+}
+
 type ServerConfig struct {
 	Port    int                         `yaml:"port"`
 	Targets map[string]*UserRolesConfig `mapstructure:"targets"`
 
-	Users []*UserConfig `mapstructure:"users"`
+	Github *GithubConfig `mapstructure:"github"`
+	Users  []*UserConfig `mapstructure:"users"`
 }
 
-func (sc *ServerConfig) NewAuthorizer() (*auth.Authorizer, error) {
+func (sc *ServerConfig) NewConfigAuthorizer() (*auth.Authorizer, error) {
 	users := map[string]*rsa.PublicKey{}
 	for _, user := range sc.Users {
 		pub, err := auth.DecodeAuthorizedKey([]byte(user.PublicKey))
 		if err != nil {
-			return nil, fmt.Errorf("Failed to decode user public key: %v, %#v", err, user.PublicKey)
+			fmt.Printf("[WARN] Failed to decode user public key: %v, %#v\n", err, user.PublicKey)
 		}
 		users[user.ID()] = pub
 	}
 	return auth.NewAuthorizer(users), nil
 }
 
+func (sc *ServerConfig) NewGithubAuthorizer() (*auth.Authorizer, error) {
+	users := map[string]*rsa.PublicKey{}
+
+	for _, user := range sc.Users {
+		if user.GithubID != "" {
+			api_endpoint := ""
+			if sc.Github != nil {
+				api_endpoint = sc.Github.APIEndpoint
+			}
+			provider_keys, err := providers.CollectGithubPublicKeys(api_endpoint, user.GithubID)
+			if err != nil {
+				return nil, err
```

Comment:
If I'm reading this right, a single github API error would result in stopping us from checking github keys for the rest of the users.  If so, it'd be better to log this error and move on to the next user.

Reply:
This should be fixed.